pkg/allocator/allocator.go

```diff
@@ -1065,6 +1065,10 @@ type AllocatorEvent struct {
 
 	// Key is the key associated with the ID
 	Key AllocatorKey
+
+	// Done is a channel that is closed when this event is processed.
+	// Optional
+	Done chan struct{}
```

Comment:
Nit. (in case you need to push again)

```suggestion
	Done chan<- struct{}
```